### PR TITLE
[tests-only][full-ci]Improve then steps usages inside acceptance tests suite `apiVersion`

### PR DESCRIPTION
**Pull request by SagarGi**
## Description
This PR adds `Then` steps to existing API test scenarios to better check the results of `When` steps.

**Suites covered:**
- `apiVersions`

## Related Issue

- Part of https://github.com/owncloud/core/issues/39512

## How Has This Been Tested?

- Locally
- CI

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiTranslation-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/35132/82

**Review by kiranparajuli589 — COMMENTED** [review]
lgtm :+1:

**Review by amrita-shrestha — COMMENTED** [review]
https://github.com/owncloud/core/pull/39932/files#diff-593d990175616dd8678fbf88613fa51b50f57522b2c52d94cc2b65a13819850bR155
` Given user "Alice" creates folder "/test" using the WebDAV API` replace with 
` Given user "Alice" has created folder "/test"`

File: tests/acceptance/features/apiVersions/fileVersionsSharingToShares.feature [hunk 1]
```diff
@@ -20,13 +20,14 @@ Feature: dav-versions
...
+    And user "Alice" has created a share with settings
       | path        | /davtest.txt |
       | shareType   | user         |
       | shareWith   | Brian        |
       | permissions | read         |
-    And user "Brian" accepts share "/davtest.txt" offered by user "Alice" using the sharing API
-    Then the version folder of fileId "<<FILEID>>" for user "Brian" should contain "1" element
+    When user "Brian" accepts share "/davtest.txt" offered by user "Alice" using the sharing API
+    Then the HTTP status code should be "200"
+    And the version folder of fileId "<<FILEID>>" for user "Brian" should contain "1" element
```
Comment (amrita-shrestha) [inline tests/acceptance/features/apiVersions/fileVersionsSharingToShares.feature:30]: In this scenario `Scenario: User can access meta folder of a file which is owned by somebody else but shared with that user`, doesn't access meta folder step should be in When?

Comment (SagarGi) [reply tests/acceptance/features/apiVersions/fileVersionsSharingToShares.feature:30]: @amrita-shrestha  there is no specidifed steps to access the meta folder. i think it is asserted in then step. also you can look into old scenario for that.

Comment (amrita-shrestha) [reply tests/acceptance/features/apiVersions/fileVersionsSharingToShares.feature:30]: @SagarGi I know, there is not any up to now. In this issue, we are shifting the main step to the **When** step. I think we can add a new step for that.  Is there any other way? 
@SwikritiT and @saw-jan

**Review by kiranparajuli589 — COMMENTED** [review]
lgtm :+1:

**Review by phil-davis — APPROVED** [review]

**Pull request merged**